core/gui/src/app/workspace/service/workflow-result-export/workflow-result-export.service.ts

```diff
@@ -100,4 +156,45 @@ export class WorkflowResultExportService {
         });
       });
   }
+
+  /**
+   * Convert the results array into CSV format and trigger download.
+   */
+  private downloadResultsAsCSV(results: any[], operatorId: string): void {
+    // Extract all unique keys from the results
+    const allKeys = new Set<string>();
+    results.forEach(record => {
+      Object.keys(record).forEach(key => allKeys.add(key));
+    });
+    const headers = Array.from(allKeys);
+
+    // Build CSV content
+    let csvContent = "";
+    // Add headers
+    csvContent += headers.join(",") + "\n";
+
+    // Add data rows
+    results.forEach(record => {
```

Comment:
You implemented the logic of converting the results to a csv file, but it is better to use a library to handle this. Please take a look at `/Users/kunwoopark/workspace/texera/core/amber/src/main/scala/edu/uci/ics/texera/web/service/ResultExportService.scala`. It has a backend code that is saving the results as a csv file to the dataset. 

Also, the backend has all the results of the operators, so I think it is better for the backend to do the conversion.

Reply:
As per our discussion, I added a library (papaparse) to parse it in the frontend but using library.